recipes/oatpp-libressl/all/conanfile.py

```diff
@@ -0,0 +1,93 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class OatppLibresslConan(ConanFile):
+    name = "oatpp-libressl"
+    license = "Apache-2.0"
+    homepage = "https://github.com/oatpp/oatpp-libressl"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "oat++ libressl library"
+    topics = ("conan", "oat++", "oatpp", "libressl")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake", "cmake_find_package", "pkg_config"
+    exports_sources = "CMakeLists.txt", "patches/**"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("oatpp-libressl can not be built as shared library on Windows")
+
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+            raise ConanInvalidConfiguration("oatpp-libressl requires GCC >=5")
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/20200517")
```

Comment:
why?

Reply:
Because was problem with finding libressl library on windows and macos

Reply:
This was why I asked (in a different pr) about if it was supposed to be msys or pkc_config (this could be changed I suspect)

Reply:
It's odd. msys2 shouldn't be required, even for this workaround

Anyway, https://github.com/conan-io/conan-center-index/pull/2933 was merge few minutes ago, therefore we can modify this recipe in a PR to avoid patch and remove this build requirement.

Reply:
@prince-chrismc upstream doesn't use at all pkg-config files, even if there is a `include(FindPkgConfig`

Reply:
Hmmm, I guess the logic in the custom find module must have picked it up in somewhere because this was enabled 🤔

Reply:
exactly

Reply:
Hi, do I have to change something?

Reply:
I've submitted https://github.com/conan-io/conan-center-index/pull/3544

Reply:
Ok, thank you